bin/hipcc

```diff
@@ -214,6 +214,7 @@ if ($HIP_PLATFORM eq "clang") {
     if (not $isWindows) {
       $HIPLDFLAGS .= " -Wl,--rpath-link=$HIP_LIB_PATH";
       $HIPLDFLAGS .= " -lhip_hcc";
+      $HIPLDFLAGS .= " -Wl,--enable-new-dtags -Wl,--rpath=$ROCM_PATH/lib:$ROCM_PATH/lib64:$ROCM_PATH/hip/lib ";
```

Comment:
What if users specified their own rpaths? Will they be overridden by the ones specified here?

Reply:
The user specified path should get appended to the these paths. The above paths helps in finding ROCM_PATH libraries. 
Also to note, we are adding RUNPATH so this can be overwritten by LD_LIBRARY_PATH if required by the user.